src/data/services/segment/utils.test.js

```diff
@@ -9,7 +9,7 @@ const courseId = 'a-fake-course-id';
 const category = 'AFakeCategory';
 describe('segment service utils', () => {
   beforeAll(() => {
-    global.window = Object.create(window);
+    global.window ??= Object.create(window);
```

Comment:
Why this statement?  The goal here is explicitly to replace the global window

Reply:
That was to resolve `TypeError: Cannot redefine property: window`. This appeared to be a [bug](https://github.com/nodejs/node/issues/47563) which was fixed by [#46615](https://github.com/nodejs/node/pull/46615) and back-ported to node v18.15 however it is still failing for node v18.16. For now I have pinned the node version to v18.15.